src/org/zaproxy/zap/extension/ascanrulesAlpha/AbstractAppFilePlugin.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+
+import org.parosproxy.paros.Constant;
+
+import java.io.IOException;
+
+public abstract class AbstractAppFilePlugin extends AbstractAppPlugin {
+
+	private static final Logger LOG = Logger.getLogger(AbstractAppFilePlugin.class);
+	private final String filename;
+	private final String MESSAGE_PREFIX;
+
+	protected AbstractAppFilePlugin(String filename, String MESSAGE_PREFIX) {
+		this.filename = filename;
+		this.MESSAGE_PREFIX = MESSAGE_PREFIX;
+	}
+	
+	private String getOtherInfo() {
+                return Constant.messages.getString(MESSAGE_PREFIX + "otherinfo");
+        }
+	
+	public void scan() {
+
+                // Check if the user stopped things. One request per URL so check before
+                // sending the request
+                if (isStop()) {
+                        if (LOG.isDebugEnabled()) {
+                                LOG.debug("Scanner " + getName() + " Stopping.");
+                        }
+                        return;
+                }
+
+                HttpMessage newRequest = getNewMsg();
+                newRequest.getRequestHeader().setMethod(HttpRequestHeader.GET);
+                URI baseUri = getBaseMsg().getRequestHeader().getURI();
+                URI elmahUri = null;
```

Comment:
elmah?

Reply:
Oh now I see that in `TraceAxdScanner.java` I made this same mistake.... guess we know what it was modeled on :)